src/helperFunctions/docker.py

```diff
@@ -1,48 +1,66 @@
+    All remaining keyword args are passed to `docker.containers.run`.
+
+    :param image: The name of the docker image
+    :param logging_label: Label used for logging
+    :param timeout: Timeout after which the execution is canceled
+    :param combine_stderr_stdout: Whether to combine stderr and stdout or not
+
+    :return: A subprocess.CompletedProcess instance for the command ran in the
+        container.
+
+    :raises docker.errors.ImageNotFound: If the docker image was not found
+    :raises requests.exceptions.ReadTimeout: If the timeout was reached
+    :raises docker.errors.APIError: If the communication with docker fails
     '''
-    container = None
+    # TODO verify that bind mounts in kwargs["mounts"] only contain files in docker-mount-base-dir
+    # If they don't just copy them to docker-mount-base-dir and change the mounts
+
+    client = docker.client.from_env()
+    kwargs.setdefault('detach', True)
+
     try:
-        kwargs = {'mounts': [Mount(*mount, read_only=False, type='bind')]} if mount else {}
-        client = docker.from_env()
-        container = client.containers.run(image, command=command, network_disabled=True, detach=True, privileged=privileged, **kwargs)
-        container.wait(timeout=timeout)
-        return container.logs(stderr=include_stderr).decode()
+        container = client.containers.run(image, **kwargs)
+    except (ImageNotFound, APIError):
+        logging.warning(f'[{logging_label}]: encountered docker error while processing')
+        raise
+
+    try:
+        response = container.wait(timeout=timeout)
+        exit_code = response['StatusCode']
+        stdout = container.logs(stdout=True, stderr=False).decode() if not combine_stderr_stdout else container.logs(stdout=True, stderr=True).decode()
+        stderr = container.logs(stdout=False, stderr=True).decode() if not combine_stderr_stdout else None
     except ReadTimeout:
-        logging.warning('[{}]: timeout while processing'.format(label))
-        if reraise:
-            raise
-    except (DockerException, IOError):
-        logging.warning('[{}]: encountered process error while processing'.format(label))
-        if reraise:
-            raise
+        logging.warning(f'[{logging_label}]: timeout while processing')
+        raise
+    except APIError:
+        logging.warning(f'[{logging_label}]: encountered docker error while processing')
+        raise
     finally:
-        if container:
-            with suppress(DockerException):
-                container.stop()
+        with suppress(DockerException):
+            container.stop()
             container.remove()
+
+    # We do not know the docker entrypoint so we just insert a generic "entrypoint"
+    command = kwargs.get('command', None)
+    command_t = type(command)
+    if command_t == str:
+        args = 'entrypoint' + command
+    elif command_t == list:
+        args = ['entrypoint'] + command
+    elif command_t:
```

Comment:
This does probably not do what it was intended to do because even `bool(type(None))` is `True`. Does this maybe do what you want?
```suggestion
    elif command is not None:
```

Reply:
Whoops, I just intendet to write `else`.